packages/web-components/fast-element/src/templating/focusgroup.ts

```diff
@@ -0,0 +1,428 @@
+                                                (Math.abs(oside.offset) <
+                                                    Math.abs(offset) ||
+                                                    // same distance but more overlap
+                                                    (Math.abs(oside.offset) ===
+                                                        Math.abs(offset) &&
+                                                        oside.overlap < overlap))))
+                                    ) {
+                                        directions[opposite] = Object.assign(
+                                            {},
+                                            { index: ci },
+                                            context[axis]
+                                        );
+                                    }
+
+                                    if (
+                                        direction === "up" &&
+                                        (directions.top.offset === undefined ||
+                                            Math.abs(offset) >
+                                                Math.abs(directions.top.offset))
+                                    ) {
+                                        directions.top = Object.assign(
+                                            {},
+                                            { index: ci },
+                                            context[axis]
+                                        );
+                                    }
+
+                                    if (
+                                        direction === "down" &&
+                                        (directions.bottom.offset === undefined ||
+                                            Math.abs(offset) >
+                                                Math.abs(directions.bottom.offset))
+                                    ) {
+                                        directions.bottom = Object.assign(
+                                            {},
+                                            { index: ci },
+                                            context[axis]
+                                        );
+                                    }
+                                }
+                            });
+                        }
+                        return directions;
+                    },
+                    {
+                        up: { index },
+                        right: { index },
+                        down: { index },
+                        left: { index },
+                        top: { index },
+                        bottom: { index },
+                    }
+                )
+            );
+    }
+
+    /**
+     * keydown handler for the parent element
+     * @param event - keyboard event
+     * @returns - should bubble
+     * @public
+     */
+    public handleKeydown(event: KeyboardEvent): boolean {
```

Comment:
How does returning a bool here work?  The event listener is hooked up via addEventListener and not a template binding and the bool only works in the latter case.  Shouldn't we always mark the event with preventDefault() when we move focus as a result?